### DISCUSSION TITLE
How come you can't update a state inside a Modal.confirm? And how do you stop the modal from auto-closing after clicking `OK` button?

### DISCUSSION THREAD

**Discussion by dmdum** [ant-design/ant-design#43849] [Q&A] [answered]
I'm using the `Modal.confirm()`. I'm passing a component as it's content. The component needs to query for some data. Before the component finishes its query, I would like to disable the modal's `Ok` button to stop unwanted function calls. I'm passing a `setState()` function to indicate if the modal has finished querying.

This is my modal:
```
const [loading, setLoading] = useState(true);

confirm({
   . . .
   content: <Component setLoading={setLoading}/>
   okButtonProps: { disabled: loading }
})
```
I am setting the state value to `false` after the component finishes querying. However, the `Ok` button is still disabled.

I'm also confused what the documentation means about this.
![image](https://github.com/ant-design/ant-design/assets/95848438/d96b035f-7767-4e6f-8f58-37055f35f60d)
I'm pretty sure I'm misinterpreting this, I tried returning true on my `onOk` function what the modal still closes...

**Comment by zombieJ** [comment, marked as answer]
https://ant.design/components/modal/#components-modal-demo-manual

**Reply by dmdum** [reply]
THANK YOU!
